content/docs/rfcs/75/README.md

```diff
@@ -0,0 +1,173 @@
+
+# Background / Rationale / Motivation
+
+Nodes who want to synchronize with each other should be able to do so with minimal amount of communication and bandwidth usage. Existing methods present in Waku Store protocol are based on full message delivery using time range query where there is no way to know apart from a given timestamp range if other messages are also missing, making them unreliable and bandwidth consuming. Node that wants to synchronize sends a request to the peer node, and in return peer node send the response to the requesting node. After a brief low-bandwidth consuming back and flow exchange, resulting will be a list values (message_hashes) that the requesting node needs to sync with the peer node. These hashes can be used by other parts of the Store protocol to make sure the related full messages are wired to the requesting node and stored in the Store, making the requesting node fully synchronized with the peer node.
+
+A detailed view in the process can be tracked using the research [issue](https://github.com/waku-org/research/issues/62).
+
+# Theory / Semantics
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “NOT RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+Consider a request-response protocol with two roles: a client and a server.
+Client and Server both MUST be peers supporting the Waku Sync protocol. There is no other eligibility for the node at this point in time to be a client or a server who can synchronize with each other.
+
+# Wire Format Specification / Syntax
+
+```
+
+    syntax = "proto3";
+
+    package prollytree;
+
+    message Node {
+        bytes data = 1;              // The data stored in the node
+        int64 timestamp = 2;          // Timestamp or other unique identifier
+        bytes node_hash = 3;         // Hash of the node
+        int32 level = 4;              // Level of the node in the tree
+        bytes merkel_hash = 5;       // Merkel hash of the node
+        bool is_tail = 6;             // Flag indicating if it's a tail node
+
+        // Relationships (IDs or indices of related nodes)
+        // Note: Relationships like left, right, up, and down are typically represented using references
+        // (like indices or IDs) rather than directly embedding the objects to avoid deep nesting.
+        // In our case we can use the key of the node itself and figure out the level based on the current node level.
```

Comment:
Would `key` here be a timestamp? Wouldn't that imply that the references should be `int64`?